### PR TITLE
embree: 3.11.0 -> 3.12.0

### PR DESCRIPTION
**Pull request by siraben**
###### Motivation for this change

Addresses #99049 

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/embree/default.nix [hunk 1]
```diff
@@ -24,8 +24,14 @@ stdenv.mkDerivation rec {
     "-DEMBREE_RAY_MASK=ON"
   ];
 
+  patchPhase = ''
+    substituteInPlace common/math/math.h --replace 'defined(__MACOSX__) && !defined(__INTEL_COMPILER)' 0
+    substituteInPlace common/math/math.h --replace 'defined(__WIN32__) || defined(__FreeBSD__)' 1
```
Comment (gebner) [inline pkgs/development/libraries/embree/default.nix:29]:
```suggestion
    substituteInPlace common/math/math.h --replace 'defined(__WIN32__) || defined(__FreeBSD__)' 'defined(__WIN32__) || defined(__FreeBSD__) || defined(__MACOSX__)'
```

We have `sincosf` on linux.

File: pkgs/development/libraries/embree/default.nix [hunk 2]
```diff
@@ -24,8 +24,14 @@ stdenv.mkDerivation rec {
     "-DEMBREE_RAY_MASK=ON"
   ];
 
+  patchPhase = ''
```
Comment (gebner) [inline pkgs/development/libraries/embree/default.nix:27]: Please put this in the `postPatch` hook.  At the moment, it overrides `postPatch` and this breaks the blender build.

File: pkgs/development/libraries/embree/default.nix [hunk 3]
```diff
@@ -24,8 +24,14 @@ stdenv.mkDerivation rec {
     "-DEMBREE_RAY_MASK=ON"
   ];
 
+  postPatch = ''
```
Comment (gebner) [inline pkgs/development/libraries/embree/default.nix:27]: Now there are two postPatch :-)

Comment (siraben) [reply pkgs/development/libraries/embree/default.nix]: Oops! Thanks!

**Pull request merged**